src/main/java/com/chatapp/vartalaab/configuration/CustomWebSocketHandler.java

```diff
@@ -0,0 +1,81 @@
+package com.chatapp.vartalaab.configuration;
+
+import com.chatapp.vartalaab.dto.MessageDto;
+import com.chatapp.vartalaab.service.MessageService;
+import com.chatapp.vartalaab.service.UserSessionService;
+import com.chatapp.vartalaab.utils.GeneralUtility;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.socket.CloseStatus;
+import org.springframework.web.socket.TextMessage;
+import org.springframework.web.socket.WebSocketSession;
+import org.springframework.web.socket.handler.TextWebSocketHandler;
+
+import java.io.IOException;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+
+@Slf4j
+public class CustomWebSocketHandler extends TextWebSocketHandler {
+
+    @Autowired
+    private ConcurrentHashMap<String, WebSocketSession> sessionMap;
+
+    @Autowired
+    private UserSessionService userSessionService;
+
+    @Autowired
+    private MessageService messageService;
+
+
+    @Override
+    public void handleTextMessage(WebSocketSession session, TextMessage message) throws IOException {
+        ObjectMapper objectMapper = new ObjectMapper();
+        MessageDto messageDto = objectMapper.readValue(message.getPayload(), MessageDto.class);
+        String username = session.getPrincipal().getName();
+        messageDto.setSender(username);
+        //ACK for message sent
+        //todo: enhance ACK messages and add concurrent msg support
+        session.sendMessage(new TextMessage(GeneralUtility.AckForSent));
+        //check if receiver is online
+        String messageReceiver = messageDto.getReceiver();
+        if(userSessionService.checkIfUserIsOnline(messageReceiver)){
+            messageService.forwardTheMessageToReceiver(messageReceiver, message);
```

Comment:
I think the function name can be changed to something simpler, just a suggestion